src/filters/network.rs

```diff
@@ -152,13 +152,15 @@ impl From<&request::RequestType> for NetworkFilterMask {
 pub enum CompiledRegex {
     Compiled(Regex),
     CompiledSet(RegexSet),
+    None,
     MatchAll,
     RegexParsingError(regex::Error),
 }
 
 impl CompiledRegex {
     pub fn is_match(&self, pattern: &str) -> bool {
         match &self {
+            CompiledRegex::None => {panic!("use of CompiledRegex::None")} // No value, invalid call.
```

Comment:
This is the kind of thing I'd prefer to see behind the `debug-info` flag, so we can guarantee it won't cause crashes for users in the wild.

Can we do this?
- Rename `CompiledRegex::None` to `CompiledRegex::Discarded`
- Put `CompiledRegex::Discarded` behind a `#[cfg(feature = "debug-info")]`
- Completely remove the `CompiledRegex` in `cleanup` when `debug-info` is disabled

Reply:
In theory we can, I also though about the similar approach.
But I believe we need discarded map entries to make a better decision in future (=stop discarding the entries that we often recreate.

Reply:
> But I believe we need discarded map entries to make a better decision in future (=stop discarding the entries that we often recreate.

I think that makes sense as an approach, but I'd strongly prefer to add panics like this only when necessary. Unless you're planning to implement that strategy in this PR, let's leave it out and we can revisit later.

There could also be alternatives to this pattern that wouldn't panic (maybe we can store hashes of the discarded rules in a bloom filter or hashmap?). But we should look at that holistically.

Reply:
Changed to `Option<CompiledRegex>`.